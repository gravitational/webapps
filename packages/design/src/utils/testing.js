import React from 'react';
import {
  render as testingRender,
  screen,
  fireEvent,
} from '@testing-library/react';
import ThemeProvider from 'design/ThemeProvider';
import theme from 'design/theme';
import 'jest-styled-components';

export function render(component) {
  return testingRender(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export { screen, fireEvent, theme };
