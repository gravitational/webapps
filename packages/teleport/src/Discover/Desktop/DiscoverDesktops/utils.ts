import { Path } from 'teleport/Discover/Desktop/DiscoverDesktops/lines/Path';
import { MonotoneX } from 'teleport/Discover/Desktop/DiscoverDesktops/lines/MonotoneX';

const STROKE_WIDTH = 4;

export interface Line {
  width: number;
  height: number;
  path: string;
}

export function createLine(
  desktopServiceElement: HTMLDivElement,
  desktopElement: HTMLDivElement,
  containerElement: HTMLDivElement
): Line {
  if (!desktopElement || !desktopServiceElement || !containerElement) {
    return null;
  }

  const desktopServiceRect = desktopServiceElement.getBoundingClientRect();
  const desktopRect = desktopElement.getBoundingClientRect();
  const containerRect = containerElement.getBoundingClientRect();

  const distance = desktopRect.left - desktopServiceRect.right;

  const path = new Path();
  const line = new MonotoneX(path);

  line.lineStart();

  const desktopLinePosition =
    desktopRect.top - containerRect.top + desktopRect.height / 2 - 1;
  const desktopServiceLinePosition =
    desktopServiceRect.top - containerRect.top + desktopServiceRect.height / 2;

  line.point(0, desktopServiceLinePosition - STROKE_WIDTH * 2);
  line.point(40, desktopServiceLinePosition - STROKE_WIDTH * 2);
  line.point(distance - 10, desktopLinePosition + STROKE_WIDTH / 2);
  line.point(distance, desktopLinePosition + STROKE_WIDTH / 2);

  line.lineEnd();

  return {
    width: distance,
    height: containerRect.height,
    path: path.path,
  };
}
