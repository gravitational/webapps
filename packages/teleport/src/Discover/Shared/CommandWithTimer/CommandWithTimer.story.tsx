/**
 * Copyright 2022 Gravitational, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import { CommandWithTimer } from './CommandWithTimer';

export default {
  title: 'Teleport/Discover/Shared/CommandWithTimer',
};

export const DefaultPolling = () => (
  <CommandWithTimer {...props} poll={{ state: 'polling' }} />
);

export const DefaultPollingSuccess = () => (
  <CommandWithTimer {...props} poll={{ state: 'success' }} />
);

export const DefaultPollingError = () => (
  <CommandWithTimer
    {...props}
    poll={{
      state: 'error',
      error: { reasonContents: [<>error reason 1</>, <>error reason 2</>] },
    }}
  />
);

export const CustomPolling = () => (
  <CommandWithTimer
    {...props}
    poll={{ state: 'polling', customStateDesc: 'custom polling text' }}
    header={<div>Custom Header Component</div>}
  />
);

export const CustomPollingSuccess = () => (
  <CommandWithTimer
    {...props}
    poll={{ state: 'success', customStateDesc: 'custom polling success text' }}
  />
);

export const CustomPollingError = () => (
  <CommandWithTimer
    {...props}
    poll={{
      state: 'error',
      error: {
        reasonContents: [<>error reason 1</>, <>error reason 2</>],
        customErrContent: <>custom error content</>,
      },
    }}
  />
);

const props = {
  command: 'some kind of command',
  pollingTimeout: 100000,
};
