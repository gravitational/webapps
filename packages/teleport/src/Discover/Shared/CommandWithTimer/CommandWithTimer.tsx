// Copyright 2022 Gravitational, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import styled from 'styled-components';
import { Text, Box } from 'design';
import * as Icons from 'design/Icon';

import { TextSelectCopyMulti } from 'teleport/components/TextSelectCopy';
import { Timeout } from 'teleport/Discover/Shared/Timeout';
import { TextIcon } from 'teleport/Discover/Shared';

export const CommandWithTimer = ({
  command,
  poll,
  pollingTimeout,
  header,
}: Props) => {
  return (
    <PollBox p={3} borderRadius={3} pollState={poll.state}>
      {header || <Text bold>Command</Text>}
      <Box mt={2} mb={1}>
        <TextSelectCopyMulti lines={[{ text: command }]} />
      </Box>
      {poll.state === 'polling' && (
        <TextIcon
          css={`
            white-space: pre;
          `}
        >
          <Icons.Restore fontSize={4} />
          <Timeout
            timeout={pollingTimeout}
            message={`${
              poll.customStateDesc || 'Waiting for Teleport Service'
            }  |  `}
          />
        </TextIcon>
      )}
      {poll.state === 'success' && (
        <TextIcon>
          <Icons.CircleCheck ml={1} color="success" />
          {poll.customStateDesc ||
            'The Teleport Service successfully join this Teleport cluster'}
        </TextIcon>
      )}
      {poll.state === 'error' && (
        <Box>
          <TextIcon>
            <Icons.Warning ml={1} color="danger" />
            {poll.error.customErrContent ||
              'We could not detect the Teleport Service you were trying to add'}
          </TextIcon>
          <Text bold mt={4}>
            Possible reasons
          </Text>
          <ul
            css={`
              margin-top: 6px;
              margin-bottom: 0;
            `}
          >
            {poll.error.reasonContents.map((content, index) => (
              <li key={index}>{content}</li>
            ))}
          </ul>
        </Box>
      )}
    </PollBox>
  );
};

type PollError = {
  reasonContents: React.ReactNode[];
  customErrContent?: React.ReactNode;
};

export type PollState = 'polling' | 'success' | 'error';

export type Poll = {
  state: 'polling' | 'success' | 'error';
  customStateDesc?: string;
  // error only needs to be defined when
  // poll state is 'error'.
  error?: PollError;
};

export const PollBox = styled(Box)`
  max-width: 800px;
  background-color: rgba(255, 255, 255, 0.05);
  padding: ${props => `${props.theme.space[3]}px`};
  border-radius: ${props => `${props.theme.space[2]}px`};
  border: 2px solid
    ${props => {
      switch (props.pollState) {
        case 'error':
          return props.theme.colors.danger;
        case 'success':
          return props.theme.colors.success;
        default:
          // polling
          return '#2F3659';
      }
    }};
`;

export type Props = {
  command: string;
  poll: Poll;
  pollingTimeout: number;
  header?: React.ReactNode;
};
