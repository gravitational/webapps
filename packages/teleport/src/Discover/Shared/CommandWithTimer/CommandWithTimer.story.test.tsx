/**
 * Copyright 2022 Gravitational, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { render } from 'design/utils/testing';

import * as stories from './CommandWithTimer.story';

test('render default polling', async () => {
  const { container } = render(<stories.DefaultPolling />);
  expect(container).toMatchSnapshot();
});

test('render default polling success', async () => {
  const { container } = render(<stories.DefaultPollingSuccess />);
  expect(container).toMatchSnapshot();
});

test('render default polling error', async () => {
  const { container } = render(<stories.DefaultPollingError />);
  expect(container).toMatchSnapshot();
});

test('render custom polling', async () => {
  const { container } = render(<stories.CustomPolling />);
  expect(container).toMatchSnapshot();
});

test('render custom polling success', async () => {
  const { container } = render(<stories.CustomPollingSuccess />);
  expect(container).toMatchSnapshot();
});

test('render custom polling error', async () => {
  const { container } = render(<stories.CustomPollingError />);
  expect(container).toMatchSnapshot();
});
