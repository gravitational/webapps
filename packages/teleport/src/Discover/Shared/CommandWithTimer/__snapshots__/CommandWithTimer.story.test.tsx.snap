// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render custom polling 1`] = `
.c1 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c4 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c6 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c8 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c8:active {
  opacity: 0.56;
}

.c8:hover,
.c8:focus {
  background: #2C3A73;
}

.c8:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c11 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c13 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
  font-size: 18px;
}

.c5 {
  box-sizing: border-box;
  display: flex;
}

.c10 .icon-check {
  display: none;
}

.c10 .icon-copy {
  display: block;
}

.c10.copied .icon-check {
  display: block;
}

.c10.copied .icon-copy {
  display: none;
}

.c9 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c3 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c2 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c7 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #2F3659;
}

.c12 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
  white-space: pre;
}

.c12 .icon {
  margin-right: 8px;
}

<div>
  <div
    class="c0"
  >
    <div>
      Custom Header Component
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c5"
              >
                <div
                  class="c6"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c7"
              >
                <button
                  class="c8 c9"
                  kind="secondary"
                >
                  <div
                    class="c10 icon-container"
                  >
                    <span
                      class="c11 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c11 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <span
        class="c13 icon icon-restore "
        color="light"
        font-size="4"
      />
      <span>
        custom polling text  |  
         
        00
        :
        00
        
      </span>
    </div>
  </div>
</div>
`;

exports[`render custom polling error 1`] = `
.c2 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c5 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c7 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c13 {
  box-sizing: border-box;
}

.c9 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c9:active {
  opacity: 0.56;
}

.c9:hover,
.c9:focus {
  background: #2C3A73;
}

.c9:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c12 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c15 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: 4px;
  color: #f50057;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
}

.c16 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
  margin-top: 24px;
}

.c6 {
  box-sizing: border-box;
  display: flex;
}

.c11 .icon-check {
  display: none;
}

.c11 .icon-copy {
  display: block;
}

.c11.copied .icon-check {
  display: block;
}

.c11.copied .icon-copy {
  display: none;
}

.c10 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c4 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c8 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c14 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
}

.c14 .icon {
  margin-right: 8px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #f50057;
}

.c17 {
  margin-top: 6px;
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Command
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c9 c10"
                  kind="secondary"
                >
                  <div
                    class="c11 icon-container"
                  >
                    <span
                      class="c12 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c12 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <span
          class="c15 icon icon-warning "
          color="danger"
        />
        custom error content
      </div>
      <div
        class="c16"
      >
        Possible reasons
      </div>
      <ul
        class="c17"
      >
        <li>
          error reason 1
        </li>
        <li>
          error reason 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`render custom polling success 1`] = `
.c2 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c5 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c7 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c9 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c9:active {
  opacity: 0.56;
}

.c9:hover,
.c9:focus {
  background: #2C3A73;
}

.c9:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c12 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c14 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: 4px;
  color: #00bfa5;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
}

.c6 {
  box-sizing: border-box;
  display: flex;
}

.c11 .icon-check {
  display: none;
}

.c11 .icon-copy {
  display: block;
}

.c11.copied .icon-check {
  display: block;
}

.c11.copied .icon-copy {
  display: none;
}

.c10 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c4 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c8 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
}

.c13 .icon {
  margin-right: 8px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #00bfa5;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Command
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c9 c10"
                  kind="secondary"
                >
                  <div
                    class="c11 icon-container"
                  >
                    <span
                      class="c12 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c12 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <span
        class="c14 icon icon-checkmark-circle "
        color="success"
      />
      custom polling success text
    </div>
  </div>
</div>
`;

exports[`render default polling 1`] = `
.c2 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c5 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c7 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c9 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c9:active {
  opacity: 0.56;
}

.c9:hover,
.c9:focus {
  background: #2C3A73;
}

.c9:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c12 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c14 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
  font-size: 18px;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
}

.c6 {
  box-sizing: border-box;
  display: flex;
}

.c11 .icon-check {
  display: none;
}

.c11 .icon-copy {
  display: block;
}

.c11.copied .icon-check {
  display: block;
}

.c11.copied .icon-copy {
  display: none;
}

.c10 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c4 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c8 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #2F3659;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
  white-space: pre;
}

.c13 .icon {
  margin-right: 8px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Command
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c9 c10"
                  kind="secondary"
                >
                  <div
                    class="c11 icon-container"
                  >
                    <span
                      class="c12 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c12 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <span
        class="c14 icon icon-restore "
        color="light"
        font-size="4"
      />
      <span>
        Waiting for Teleport Service  |  
         
        00
        :
        00
        
      </span>
    </div>
  </div>
</div>
`;

exports[`render default polling error 1`] = `
.c2 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c5 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c7 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c13 {
  box-sizing: border-box;
}

.c9 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c9:active {
  opacity: 0.56;
}

.c9:hover,
.c9:focus {
  background: #2C3A73;
}

.c9:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c12 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c15 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: 4px;
  color: #f50057;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
}

.c16 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
  margin-top: 24px;
}

.c6 {
  box-sizing: border-box;
  display: flex;
}

.c11 .icon-check {
  display: none;
}

.c11 .icon-copy {
  display: block;
}

.c11.copied .icon-check {
  display: block;
}

.c11.copied .icon-copy {
  display: none;
}

.c10 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c4 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c8 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c14 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
}

.c14 .icon {
  margin-right: 8px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #f50057;
}

.c17 {
  margin-top: 6px;
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Command
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c9 c10"
                  kind="secondary"
                >
                  <div
                    class="c11 icon-container"
                  >
                    <span
                      class="c12 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c12 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      >
        <span
          class="c15 icon icon-warning "
          color="danger"
        />
        We could not detect the Teleport Service you were trying to add
      </div>
      <div
        class="c16"
      >
        Possible reasons
      </div>
      <ul
        class="c17"
      >
        <li>
          error reason 1
        </li>
        <li>
          error reason 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`render default polling success 1`] = `
.c2 {
  box-sizing: border-box;
  margin-bottom: 4px;
  margin-top: 8px;
}

.c5 {
  box-sizing: border-box;
  padding-bottom: 0px;
  padding-top: 8px;
}

.c7 {
  box-sizing: border-box;
  margin-right: 4px;
}

.c9 {
  line-height: 1.5;
  margin: 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: inherit;
  font-weight: 600;
  outline: none;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: all 0.3s;
  -webkit-font-smoothing: antialiased;
  background: #222C59;
  color: rgba(255,255,255,0.87);
  min-height: 32px;
  font-size: 12px;
  padding: 0px 24px;
}

.c9:active {
  opacity: 0.56;
}

.c9:hover,
.c9:focus {
  background: #2C3A73;
}

.c9:disabled {
  background: rgba(255,255,255,0.12);
  color: rgba(255,255,255,0.3);
}

.c12 {
  display: inline-block;
  transition: color 0.3s;
  color: #FFFFFF;
}

.c14 {
  display: inline-block;
  transition: color 0.3s;
  margin-left: 4px;
  color: #00bfa5;
}

.c1 {
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 600;
  margin: 0px;
}

.c6 {
  box-sizing: border-box;
  display: flex;
}

.c11 .icon-check {
  display: none;
}

.c11 .icon-copy {
  display: block;
}

.c11.copied .icon-check {
  display: block;
}

.c11.copied .icon-copy {
  display: none;
}

.c10 {
  height: 28px;
  width: 28px;
  border-radius: 20px;
  min-height: auto;
  padding: 0;
  margin-top: -4px;
}

.c4 {
  box-sizing: border-box;
  margin-right: 4px;
  white-space: pre;
  word-break: break-all;
  font-size: 12px;
  font-family: "Droid Sans Mono","monospace",monospace,"Droid Sans Fallback";
  overflow: scroll;
  line-height: 20px;
}

.c3 {
  box-sizing: border-box;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 48px;
  padding-top: 8px;
  background-color: #010B1C;
  border-radius: 4px;
  position: relative;
}

.c8 {
  box-sizing: border-box;
  padding-right: 16px;
  position: absolute;
  right: 0px;
}

.c13 {
  overflow: hidden;
  text-overflow: ellipsis;
  margin: 0px;
  display: flex;
  align-items: center;
}

.c13 .icon {
  margin-right: 8px;
}

.c0 {
  box-sizing: border-box;
  padding: 16px;
  border-radius: 8px;
  max-width: 800px;
  background-color: rgba(255,255,255,0.05);
  padding: 16px;
  border-radius: 8px;
  border: 2px solid #00bfa5;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Command
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c6"
              >
                <div
                  class="c7"
                >
                  $
                </div>
                <div
                  class="text-to-copy"
                >
                  some kind of command
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c9 c10"
                  kind="secondary"
                >
                  <div
                    class="c11 icon-container"
                  >
                    <span
                      class="c12 icon icon-copy "
                      color="light"
                      data-testid="btn-copy"
                    />
                    <span
                      class="c12 icon icon-check "
                      color="light"
                      data-testid="btn-check"
                    />
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <span
        class="c14 icon icon-checkmark-circle "
        color="success"
      />
      The Teleport Service successfully join this Teleport cluster
    </div>
  </div>
</div>
`;
